applications/ShapeOptimizationApplication/python_scripts/model_part_controller_factory.py

```diff
@@ -88,6 +82,14 @@ def GetDampingRegions(self):
             self.__IdentifyDampingRegions()
         return self.damping_regions
 
+    # --------------------------------------------------------------------------
+    def __IsOptimizationModelPartAlreadyImported(self):
+        if self.optimization_mdpa.NumberOfNodes()>0:
+            self.__CheckIfDomainSizeIsSet()
+            return True
+        else:
+            return False
+
     # --------------------------------------------------------------------------
     def __CheckIfDomainSizeIsSet(self):
         if self.optimization_mdpa.ProcessInfo.GetValue(DOMAIN_SIZE) == 0:
```

Comment:
This initializes the DOMAIN_SIZE in the ProcessInfo

Could you try `self.optimization_mdpa.ProcessInfo.Has(DOMAIN_SIZE)`?

Reply:
Does not work.